TeamNewPipe/NewPipe
# Improve screen rotation handling in Open action menu

#### What is it?

#### Description of the changes in your PR

Currently when sharing to NewPipe (when set to `Always ask`), the Open action menu dialog would rotate away on orientation change (since `finish()` would be called by onDismissListener during dialog cleanup thus preventing `RouterActivity` to be recreated when screen rotates).  This PR tries to improve that by making the dialogs stay across screen rotations.

~~_Known issue:_ Ongoing network requests, however, won't survive config change (which would somehow require keeping track of the `Observable`s across recreation of activity).  If user selects Add to Playlist or Download and rotates away while stream info is being fetched, the network request underway would be dropped and `RouterActivity` would start over again when recreated on orientation change.  That would (unfortunately) be another PR for another day.~~ Addressed this using a retained fragment the other day

##### Misc changes
- Fixes a bug on older Android where the Open action dialog won't appear the first time on cold start
- Shows on Download action as well the toast that asks the user to wait
- When fetching stream info (for Add to Playlist and Download), the transparent activity no longer blocks touch events and inadvertently locks UI thus giving the impression that NewPipe freezes

##### To test:
- Make sure `Settings` -> `Video & audio` -> `Preferred 'open' action` is set to `Always ask` (the default setting)
- Share a (YouTube) URL to NewPipe (Debug apk) (say, from a browser)
- Happy screen rotating :)

#### Fixes the following issue(s)

- Fixes #3129

#### APK testing


The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
